### PR TITLE
Update Dependencies & base Python version

### LINKED ISSUE #22: Provide error message when "unexpected exception" occurs
We had a case where the AI Assistant could not connect to its target Vantiq server (due to networking issues).  The error was:

```
VantiqException: code: io.vantiq.python.operationerror, message=Unexpected error during selectOne operation.
```

We need to include the actual error message to make diagnosis easier.

### LINKED ISSUE #23: update dependencies to match extpsdk & pythonexecconnector

### PR DESCRIPTION
**Pull request by fhcarter**
Fixes #23
Fixes #22

Update dependencies to more modern work.  Updated base Python to 3.10 to match what (will) happen in the extension source's repo.  3.8 is likely to continue to work, but we don't test it -- that version's pretty old.

Updated requirements to fix dependabot-reported issues.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -100,8 +95,6 @@ pytest-metadata==3.0.0
     # via pytest-html
 pytest-timeout==2.1.0
     # via -r requirements-build.in
-pywin32-ctypes==0.2.2
```
Comment (jcguion) [inline requirements.txt:103]: Maybe this should be changed as `pywin32-ctypes==0.2.2 ;  platform_system == 'Windows'`, instead of being removed. We might also want to keep colorama in case it comes from a windows requirements.txt update, although it doesn't appear to be windows specific (safe to keep as is).

**Review by jcguion — APPROVED** [review]
LGTM

File: requirements.txt (hunk 1 above)
Comment (sfitts) [reply requirements.txt:103]: Python is annoying in that even if you do that in the `.in` files, when you generate on "not Windows" it still won't be included.  In addition, the transitive deps will also be removed.  I get the latter since it is hard to follow deps for something you aren't installing (though a more robust build system would have ways to solve that), but the fact that the root project is also ignored is.. a real PITA.  Overall I think the only strategy is to examine any removals (such as these) and manually remove them unless you know they are due to other changes and aren't platform specific.  Otherwise the only way to do it is to perform requirements generation on all possible platforms and merge the result (not something that is practical).

Comment (fhcarter) [reply requirements.txt:103]: I'll add those two back in with associated comments that say something about "manually" added.  Not sure that will really fix things since we regenerate the requirements.txt file when we determine it's necessary.  But perhaps that's all that's possible in this environment.

Comment (fhcarter) [reply requirements.txt:103]: @jcguion & @sfitts ... is this a reasonable solution?  If so, I'll assume it's OK, merge, and publish to pypi

Comment (jcguion) [reply requirements.txt:103]: LGTM
